sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -857,12 +855,12 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
 
     // Checks default compression codec
     checkCompressionCodec(
-      CompressionCodecName.fromConf(spark.conf.get(SQLConf.PARQUET_COMPRESSION)))
+      ParquetCompressionCodec.fromString(spark.conf.get(SQLConf.PARQUET_COMPRESSION)))
 
-    checkCompressionCodec(CompressionCodecName.UNCOMPRESSED)
-    checkCompressionCodec(CompressionCodecName.GZIP)
-    checkCompressionCodec(CompressionCodecName.SNAPPY)
-    checkCompressionCodec(CompressionCodecName.ZSTD)
+    checkCompressionCodec(ParquetCompressionCodec.UNCOMPRESSED)
```

Comment:
Unrelated to this pr, but why were only four types of Compression Codec tested here? Was the test case not modified when a new type was added?

Reply:
I tested the other compression codec, the tests failed!
It seems not supported the others yet.

Reply:
ok

Reply:
If I have time, I will try to cover these tests.

Reply:
I got it now. `lzo` is supported by cloudera Hadoop. Spark doesn't have it built-in.